mgmtd/mgmt_txn.c

```diff
@@ -2695,8 +2695,10 @@ int mgmt_txn_notify_error(struct mgmt_be_client_adapter *adapter,
 	case MGMTD_TXN_PROC_RPC:
 		rpc = txn_req->req.rpc;
 		rpc->recv_clients |= (1u << id);
-		rpc->errstr = XSTRDUP(MTYPE_MGMTD_ERR, errstr);
-
+		if (errstr) {
+			XFREE(MTYPE_MGMTD_ERR, rpc->errstr);
```

Comment:
this pattern appears ... multiple times - would it be better to offer an api for it?

Reply:
I believe it's only used in 2 places, and is only 2 lines: free old, set new, so i'd prefer to just leave the code obvious.